sim/SIM-PROTOCOL.md

```diff
@@ -443,10 +523,41 @@ stat boosts are minor actions.
 > for messages from game mods that aren't supported by the client, like rule clauses 
 > such as Sleep Clause, or other metagames with custom messages for specific scenarios. 
 
-I'll document all the message types eventually, but for now this should be
-enough to get you started. You can watch the data sent and received from
-the server on a regular connection, or look at the client source code
-for a full list of message types.
+`|-combine`
+
+> A move has been combined with another (For example: Fire Pledge).
+
+`|-waiting|SOURCE|TARGET`
+
+> The `SOURCE` Pokémon has used a move and is waiting for the `TARGET` Pokémon
+> (For example: Fire Pledge).
+
+`|-prepare|ATTACKER|MOVE|DEFENDER`
+
+> The `ATTACKER` Pokémon is preparing to use a charge `MOVE` on the `DEFENDER`
+> (For example: Dig, Fly).
+
+`|-mustrecharge|POKEMON`
+
+> The Pokémon `POKEMON` must spend the turn recharging from a previous move.
+
+`|-nothing`
+
+> A move did absolutely nothing. (For example: Splash).
```

Comment:
This is deprecated; it's now `|-activate|...|move: Splash`.

Reply:
Yeah, I noticed that on the client the `upgradeArgs`, but it still seems to be generated by the server? e.g. https://github.com/Zarel/Pokemon-Showdown/blob/master/data/mods/gen2/moves.js#L736

Should I remove this entitely, or just add a note in the documentation that it's deprecated?

Reply:
Either is up to you, but `upgradeArgs` is explicitly supposed to upgrade a deprecated message type into the modern equivalent. Anything sent by the server that's handled by `upgradeArgs` is supposed to eventually become the new standard sent by the server.

Reply:
Although with some extreme caveats that `upgradeArgs` represents "proposed" changes in the sense that I don't want to go through and fix server output _yet_.

Reply:
So marking it as "deprecated" here may be the best solution.